code/go/0chain.net/miner/miner/miner.go

```diff
@@ -148,8 +148,15 @@ func main() {
 	// TODO: put it in a better place
 	go mc.StartLFMBWorker(ctx)
 
-	gb := mc.SetupGenesisBlock(viper.GetString("server_chain.genesis_block.id"),
-		magicBlock, initStates)
+	var gb *block.Block
+
+	lfb := miner.GetLatestBlockFromSharders(ctx)
+
+	if mc.GetCurrentRound() == 0 || lfb == nil {
```

Comment:
We should only check the current round, the lfb would be nil before current round is >= 3, however we should only initialize genesis block once.

Reply:
@peterlimg updated.